### PR TITLE
WIP: Add repository tests

### PR DESCRIPTION
**Pull request by nlubisch**
### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

File: htdocs/app/Migrations/Version20171008183000.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class Version20171008183000 extends AbstractMigration
+{
+    /**
+     * Sets the engine of the table user to INNODB.
+     *
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        $this->addSql('ALTER TABLE user ENGINE = INNODB;');
```
Comment (teiling88) [inline htdocs/app/Migrations/Version20171008183000.php:22]: no innodb at the moment. Have a look into our board. https://forum.opencaching.de/index.php?topic=4762.msg61245#msg61245

File: sql/stored-proc/maintain-current.inc.php [hunk 2]
```diff
@@ -2387,19 +2387,7 @@
 sql(
     "CREATE TRIGGER `userBeforeDelete` BEFORE DELETE ON `user`
      FOR EACH ROW BEGIN
-        IF IFNULL(@allowdelete,0)=0 THEN
-            CALL error_set_allowdelete_to_delete_users();
```
Comment (following5) [inline sql/stored-proc/maintain-current.inc.php:2391]: This procedure call is intentional. It raises an SQL error if @allowdelete is not set, to get the developer's attention to the following comments. 

Please don't delete this code, or implement another way to throw some error or exception here which points the developer to an explanation on deleting user records.

Comment (following5) [reply sql/stored-proc/maintain-current.inc.php:2391]: Btw, even if the procedure call would have been obsolete, your change would have removed the complete IF condition which prevents deleting a user if @allowdelete is not set!

File: htdocs/app/Migrations/Version20171008183000.php (hunk 1 above)
Comment (teiling88) [reply htdocs/app/Migrations/Version20171008183000.php:22]: Innodb is active since a couple of months without any issue :-)

**Comment by teiling88** [comment]
Closed during inactivity!

**Pull request closed without merging**